src/table.rs

```diff
@@ -202,6 +203,89 @@ impl Table {
             .collect()
     }
 
+    // count means including only positive values. Implement other functions like sum_group_by, mean_group_by, etc. later.
+    #[must_use]
+    pub fn count_group_by(
+        &self,
+        rows: &[usize],
+        column_types: &Arc<Vec<ColumnType>>,
+        by_column: usize,
+        by_interval: Option<u32>,
+        count_columns: &Arc<Vec<usize>>,
+    ) -> Vec<GroupCount> {
+        let rows_interval: Vec<GroupElement> = match column_types[by_column] {
+            ColumnType::DateTime => {
+                if let Some(by_interval) = by_interval {
+                    convert_time_intervals(
+                        self.columns
+                            .get(by_column)
+                            .expect("time column should exist"),
```

Comment:
`count_group_by` is a public method, and thus we have to assume that its argument, `by_column`, may take any `usize` value.

Reply:
Added validation of arguments, `by_column` and `count_columns`.